pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/src/main/java/org/apache/pinot/plugin/stream/kafka20/KafkaConfigBackwardCompatibleUtils.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.plugin.stream.kafka20;
+
+import java.util.Map;
+import org.apache.pinot.spi.stream.StreamConfig;
+
+
+public class KafkaConfigBackwardCompatibleUtils {
+  private KafkaConfigBackwardCompatibleUtils() {
+  }
+
+  public static final String KAFKA_COMMON_PACKAGE_PREFIX = "org.apache.kafka.common";
+  public static final String PINOT_SHADED_PACKAGE_PREFIX = "org.apache.pinot.shaded.";
+
+  /**
+   * Handle the stream config to replace the Kafka common package with the shaded version if needed.
+   */
+  public static void handleStreamConfig(StreamConfig streamConfig) {
+    Map<String, String> streamConfigMap = streamConfig.getStreamConfigsMap();
+    for (Map.Entry<String, String> entry : streamConfigMap.entrySet()) {
+      String[] valueParts = entry.getValue().split(" ");
+      boolean updated = false;
+      for (int i = 0; i < valueParts.length; i++) {
+        if (valueParts[i].startsWith(KAFKA_COMMON_PACKAGE_PREFIX)) {
+          try {
+            Class.forName(valueParts[i]);
+          } catch (ClassNotFoundException e) {
+            // If not, replace the class with the shaded version
+            valueParts[i] = PINOT_SHADED_PACKAGE_PREFIX + valueParts[i];
```

Comment:
Let's check if the shaded one can be found here

Reply:
done